github_metrics/pr_story.py

```
@@ -0,0 +1,804 @@
+
+from simple_logger.logger import get_logger
+
+from github_metrics.database import DatabaseManager
+
+LOGGER = get_logger(name="github_metrics.pr_story")
+
+# Event grouping window - events within this duration are considered parallel
+GROUPING_WINDOW_SECONDS = 60
+
+
+def _parse_payload(payload: Any) -> dict[str, Any]:
```

Ruff (0.14.6)

Dynamically typed expressions (typing.Any) are disallowed in `payload`

(ANN401)